File: Bookworm Backend Dot net/Models/Product.cs
using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;
using System.Collections.Generic;
using System;
using Microsoft.EntityFrameworkCore;

namespace Bookworm.Models
{
    [Table("product_master")]
    [Index(nameof(Isbn), IsUnique = true)]
    public class Product
    {
        [Key]
        [Column("product_id")]
        public int Id { get; set; }

        [StringLength(255)]
        [Column("product_author")]
        public string Author { get; set; }

        [Required]
        [Column("product_base_price", TypeName = "decimal(10, 2)")]
        public decimal BasePrice { get; set; }

        [StringLength(255)]
        [Column("english_name")]
        public string EnglishName { get; set; }

        [StringLength(255)]
        [Column("img_src")]
        public string ImageSource { get; set; }

        [Column("is_rentable")]
        public bool? IsRentable { get; set; }

        [StringLength(20)]
        [Column("product_isbn")]
        public string Isbn { get; set; }

        [Column("long_description", TypeName = "TEXT")]
        public string LongDescription { get; set; }

        [Column("min_rent_days")]
        public int? MinRentDays { get; set; }

        [Required]
        [StringLength(255)]
        [Column("product_name")]
        public string Name { get; set; }

        [Column("product_offer_price", TypeName = "decimal(10, 2)")]
        public decimal? OfferPrice { get; set; }

        [Column("rent_per_day", TypeName = "decimal(10, 2)")]
        public decimal? RentPerDay { get; set; }

        [Column("product_sp_cost", TypeName = "decimal(10, 2)")]
        public decimal? SpecialCost { get; set; }

        [StringLength(255)]
        [Column("short_description")]
        public string ShortDescription { get; set; }

        [Required]
        [Column("genre_id")]
        [ForeignKey("Genre")]
        public int GenreId { get; set; }
        public virtual Genre Genre { get; set; }

        [Required]
        [Column("language_id")]
        [ForeignKey("Language")]
        public int LanguageId { get; set; }
        public virtual Language Language { get; set; }

        [Required]
        [Column("type_id")]
        [ForeignKey("ProductType")]
        public int ProductTypeId { get; set; }
        public virtual ProductType ProductType { get; set; }

        // Navigation property for one-to-many relationship with ProductBeneficiary
        public virtual ICollection<ProductBeneficiary> ProductBeneficiaries { get; set; } = new List<ProductBeneficiary>();

        public virtual ICollection<CartDetail> CartDetails { get; set; }

    }

}

File: Bookworm Backend Spring Boot/src/main/java/com/bookworm/entities/Product.java
package com.bookworm.entities;

import jakarta.persistence.*;
import lombok.*;
import java.math.BigDecimal;
import java.time.LocalDate;
import java.util.List;
import java.util.Set;

/*
 * Represents a product in the catalog, such as a book or e-book.
 * This is a central entity linking to descriptions, pricing, genres, and more.
 * Maps to the 'products' table.
 */
@Entity
@Table(name = "product_master")
@Getter
@Setter
public class Product {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "product_id")
    private Integer id;

    @Column(name = "product_name", length = 255, nullable = false)
    private String name;

    @Column(name = "english_name", length = 255)
    private String englishName;

    @Column(name = "product_author", length = 255)
    private String author;

    @Column(name = "product_isbn", length = 20, unique = true)
    private String isbn;

    @Lob
    @Basic(fetch = FetchType.LAZY)
    @Column(name = "long_description", columnDefinition = "TEXT")
    private String longDescription;

    @Column(name = "short_description", length = 255) // Corrected from 25 to 255
    private String shortDescription;

    @Column(name = "img_src", length = 255)
    private String imageSource;

    @Column(name = "is_rentable")
    private boolean isRentable;

    @Column(name = "min_rent_days")
    private Integer minRentDays;

    @Column(name = "rent_per_day", precision = 10, scale = 2)
    private BigDecimal rentPerDay;

    @Column(name = "product_base_price", precision = 10, scale = 2, nullable = false)
    private BigDecimal basePrice;

    @Column(name = "product_offer_price", precision = 10, scale = 2)
    private BigDecimal offerPrice;

    @Column(name = "product_sp_cost", precision = 10, scale = 2)
    private BigDecimal specialCost;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "genre_id", nullable = false)
    private Genre genre;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "language_id", nullable = false)
    private Language language;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "type_id", nullable = false)
    private ProductType productType;
    
    @OneToMany(mappedBy = "product", cascade = CascadeType.ALL)
    private List<ProductBeneficiary> productBeneficiaries;


//    @OneToMany(mappedBy = "product", cascade = CascadeType.ALL, orphanRemoval = true)
//    private Set<ProductAttribute> attributes;
//
//    @OneToMany(mappedBy = "product", cascade = CascadeType.ALL, orphanRemoval = true)
//    private Set<ProductBeneficiary> beneficiaries;
//
//    // Relationships to transactional details are typically not managed via cascade from Product
//    @OneToMany(mappedBy = "product")
//    private Set<InvoiceDetail> invoiceDetails;
//
      @OneToMany(mappedBy = "product")
     private Set<CartDetail> cartDetails;
//
//    @OneToMany(mappedBy = "product")
//    private Set<RentDetail> rentDetails;

}